### PR TITLE
Improve npm/yarn scripts to run jest tests

### PR DESCRIPTION
**Pull request by PierreBesson**
Following https://github.com/jhipster/generator-jhipster/pull/8816, this simply cleans up the script and add a new script to run `jest --updateSnapshot` by itself.

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
Azure builds failed.
Maybe this script need to be updated ? https://github.com/jhipster/generator-jhipster/blob/master/test-integration/scripts/22-tests-frontend.sh

Then, all files for CICD should use test-ci right ?
For example: https://github.com/jhipster/generator-jhipster/blob/master/generators/ci-cd/templates/travis.yml.ejs#L74

**Comment by pascalgrimaud** [comment]
@PierreBesson : don't hesitate to tell me if you need help to update CICD files

File: generators/client/templates/react/package.json.ejs [hunk 1]
```diff
@@ -194,8 +194,10 @@ limitations under the License.
     <%_ } _%>
     "start": "<%= clientPackageManager %> run webpack:dev",
     "start-tls": "<%= clientPackageManager %> run webpack:dev <%= optionsForwarder %>--env.tls",
-    "test": "<%= clientPackageManager %> run lint && jest --coverage --logHeapUsage -w=2 --config src/test/javascript/jest.conf.js",
-    "test-ci": "<%= clientPackageManager %> run lint && jest --coverage --logHeapUsage --updateSnapshot -w=2 --config src/test/javascript/jest.conf.js",
+    "jest": "jest --coverage --logHeapUsage --maxWorkers=2 --config src/test/javascript/jest.conf.js",
+    "jest:update": "npm run jest <%= optionsForwarder %>--updateSnapshot",
```
Comment (vishal423) [inline generators/client/templates/react/package.json.ejs]: missed using `<%= clientPackageManager %>`

**Comment by PierreBesson** [comment]
I have updated the ci-cd pipelines to use the new test-ci script. But seriously the ci-cd subgen is a big mess and all the CI files are slightly different. It must be a nightmare to test how are you doing it @pascalgrimaud ? I have standardized on using maven/gradle all the time to run the frontend tests but now I will have to manually tests ci pipelines again to see if I didn't break anything.

We really have to refactor this part to at least have the commands for each step centralized in one place (eg. `runFrontEndTests: "./mvnw com.github.eirslett:frontend-maven-plugin:npm -Dfrontend.npm.arguments=test-ci'`is copy paster all over the place).

BTW @deepu105  does the new JDL support the CI-CD generator ?

**Comment by pascalgrimaud** [comment]
manually but no worry, I can test it :)

**Comment by pascalgrimaud** [comment]
@PierreBesson : I push directly to your branch, then do the same tests than my previous PR. Everything works well. Good job!

**Pull request merged**